File: exercism.io/csharp/robot-name/RobotName.cs
using System;
using System.Collections.Generic;

public class Robot
{
    private string _name;
    private Random r;
    public static List<string> repeatedNames = new List<string>();

    public Robot()
    {
        this.r = new Random();
        this._name = this.generate();
        repeatedNames.Add(this._name);
    }

    public string Name
    {
        get
        {
            return this._name;
        }
    }

    public void Reset()
    {
        repeatedNames.Remove(this._name);
        this._name = this.generate();
    }

    public string generate()
    {
        string code = this.generate_code();
        while (repeatedNames.Contains(code))
        {
            code = this.generate_code();
        }
        repeatedNames.Add(code);
        return code;
    }

    private string generate_code()
    {
        return "" + Char.ToUpper((char)('a' + this.r.Next(0, 25))) + Char.ToUpper((char)('a' + this.r.Next(0, 25))) + this.r.Next(100, 999);
    }
}

File: exercism.io/python/robot-name/robot_name.py
import random
import string

class Robot(object):
    def __init__(self):
        self.name = ''.join(random.choice(string.ascii_uppercase) for _ in range(2)) + ''.join(random.choice(string.digits) for _ in range(3))
    def reset(self):
        name = self.name
        while(name == self.name):
            self.name = "".join(random.choice(string.ascii_uppercase) for _ in range(2)) + "".join(random.choice(string.digits) for _ in range(3))

        